### ISSUE TITLE
Enhancement: Font size of components within a compact table

### ISSUE THREAD

**Issue by srambach** [patternfly/patternfly-design#854]
Background: A compact list reduces the font size of text within it. However, some components such as button, dropdowns, etc., have their own font size set and therefore are displayed with a larger font size.

Question: Should these components maintain their default size within a compact list, or should they automatically adjust to match the font size used in the compact list?

In the example below, the "Endpoint" link is a link button within a compact list. 
Upon inspection you can see that the button is using its own font sizing with
'--pf-c-button--FontSize: var(--pf-global--FontSize--md);'
while the table content is using a font size specific to the compact modifier.
`--pf-c-table-cell--FontSize: var(--pf-c-table--m-compact--FontSize);`

![image](https://user-images.githubusercontent.com/19825616/82248830-c748f900-9916-11ea-9cec-39824bf9d439.png)

@mcoker

**Comment by mcarrano** [comment]
This is a really good question @srambach . I agree that in the case you show, I would expect the link button to inherit the same type-size as other table elements. @mceledonia do you agree with that assessment? Can you think of cases where we would not want that?

@mcoker if we wanted to address this, is there a way for components enclosed in the table to inherit this default font size? It's difficult since just about anything can be placed in a table cell. Or would the project need to just override the default sizing for a specific element, like making the link button text smaller in this case?

**Comment by mcoker** [comment]
@mcarrano yes. Elements can "inherit" the existing font-size of their parent element, meaning if they're used in an element with a small font, the button will also be small font. That seems to me like the way we would go for "inline" type components that are used inline with other things versus components that define their own block on the page.

One option could be to make a button inherit font-size from it's parent, regardless of where the button is used. In a compact table, data-list, or even a small custom box an application creates. If a user uses the button in a parent with small font-size, the button will have a small font-size.

> is there a way for components enclosed in the table to inherit this default font size

We could do this, too, where we say buttons, dropdowns, etc used specifically in the table or data-list components (or even limit it further to use in the compact variation of these components) will inherit the font-size specified by the component. That would give us a little more control over how exactly the elements behave, but seems like it could lead to inconsistencies and confusion if buttons work this way in some components, but not all, or if they work this way in patternfly components, but not when used in custom designs.

Another option, specific to buttons, is we could make it so that the "inline link" button variation inherits the font-size of the parent. I think we probably should have done this when we added this variation, as it's supposed to flow with regular text content, like a link. That also seems to be Sarah's use case since she's using the inline link button in her screenshot.

**Comment by mcarrano** [comment]
Thanks @mcoker 

@srambach as a consumer of this, what would you expect?

**Comment by srambach** [comment]
I expected that the inline link button would inherit the font-size. 
However, I don't think I'd expect this for the normal button-y buttons. For those, if a smaller size were available as a variant (and could meet a11y requirements) I think that would be a good solution if I needed to put one into a compact table.

**Comment by mcoker** [comment]
@mcarrano fwiw, we have this issue to introduce a `$pf-global--FontSize--sm` (14px) button. https://github.com/patternfly/patternfly/issues/3116

**Comment by mcarrano** [comment]
@mcoker this is currently filed in patternfly-design, should I just move to the core repo for next steps? Do you agree that we have a path forward?

**Comment by mcoker** [comment]
@mcarrano yes that works.

> I expected that the inline link button would inherit the font-size.

Would you consider it a visual breaking change for the inline-link variation of the button to begin inheriting `font-size` from it's parent?

**Comment by mcarrano** [comment]
@mcoker OK, sending this over to core. I don't see this as a visual breaking change. More like fixing a bug as I think we would have always wanted it to behave this way if we had thought it through.
